setup.py

```diff
@@ -1,52 +1,146 @@
+def untar(archive):
+    tar = tarfile.open(archive)
+    tar.extractall()
+    tar.close()
+
+
+# attempt to run parallel make with at most 8 workers
+def cpu_count():
+    try:
+        import multiprocessing
+        return min(8, multiprocessing.cpu_count())
+    except:
+        return 1
+
+
+def get_readme():
+    with open('docs/source/README.txt', 'r') as fp:
+        return fp.read()
+
+
+# Download, unzip, configure, and make spatialindex C library from github.
+# Attempt to skip steps that may have already completed.
+def build_spatialindex(libname):
+    version = '1.8.5'
+    archive = 'spatialindex-' + version + '.tar.gz'
+    destdir = 'libspatialindex-' + version
+
+    if not os.path.exists(destdir):
+        if not os.path.exists(archive):
+            print("Downloading latest spatialindex master")
+            theurl = ('https://github.com/libspatialindex/libspatialindex'
+                      + '/archive/' + version + '.tar.gz')
+            name, hdrs = urllib.request.urlretrieve(theurl, archive)
+
+        print("Unzipping spatialindex master archive")
+        untar(archive)
+
+    root = os.path.join(os.path.dirname(os.path.abspath(__file__)), destdir)
+    shutil.copy('CMakeLists.txt', root)
+    print("making spatialindex")
+    if platform.system() == "Darwin":
+        os.environ['CFLAGS'] = "-arch x86_64 -arch i386"
+    retcode = subprocess.Popen([
+        'cmake', '.',
+        ], cwd=root).wait()
+    retcode = subprocess.Popen([
+        'make',
+        '-j',
+        str(cpu_count())
+        ], cwd=root).wait()
+    if 0 != retcode:
+        raise RuntimeError("make failed")
+
+    src = os.path.join(root, 'bin')
+    dst = os.path.join('rtree', '.libs')
+    os.mkdir(dst)
+    print("copying {} to {}".format(src, dst))
+    for f in libname:
+        shutil.copy(os.path.join(src, f), dst)
+
+
+class bdist_wheel(bdist_wheel_):
+    def run(self):
+        libname = get_libname()
+        if not os.path.exists('rtree/.libs'):
+            build_spatialindex(libname)
+        if not os.path.exists('rtree/.libs'):
+            raise RuntimeError("Unable to find shared library {lib}."
+                               .format(lib=libname))
+        bdist_wheel_.run(self)
+
+    def finalize_options(self):
+        bdist_wheel_.finalize_options(self)
+        self.plat_name_supplied = True
+        self.plat_name = get_platform()
+        self.universal = False
+        self.root_is_pure = False
+
+if __name__ == "__main__":
+    if os.name == 'nt':
+        data_files = [('Lib/site-packages/rtree',
+                      [os.environ['SPATIALINDEX_LIBRARY']
+                          if 'SPATIALINDEX_LIBRARY' in os.environ else
+                          r'D:\libspatialindex\bin\spatialindex.dll',
+                       os.environ['SPATIALINDEX_C_LIBRARY']
+                          if 'SPATIALINDEX_C_LIBRARY' in os.environ else
+                          r'D:\libspatialindex\bin\spatialindex_c.dll'])]
+    else:
```

Comment:
Why `D:` ? It is just a convention we are using?

Reply:
This is from upstream. I did not touched it.